setuptools/tests/config/test_apply_pyprojecttoml.py

```diff
@@ -320,26 +319,36 @@ def test_license_in_metadata(
     assert content_str in content
 
 
-def test_license_expression_with_bad_classifier(tmp_path):
+def test_license_classifier_with_license_expression(tmp_path):
     text = PEP639_LICENSE_EXPRESSION.rsplit("\n", 2)[0]
     pyproject = _pep621_example_project(
         tmp_path,
         "README",
         f"{text}\n    \"License :: OSI Approved :: MIT License\"\n]",
     )
-    msg = "License classifier are deprecated(?:.|\n)*'License :: OSI Approved :: MIT License'"
-    with pytest.raises(SetuptoolsDeprecationWarning, match=msg):
+    msg = "License classifiers have been superseded by license expressions"
+    with pytest.raises(InvalidConfigError, match=msg) as exc:
         pyprojecttoml.apply_configuration(makedist(tmp_path), pyproject)
 
-    with warnings.catch_warnings():
-        warnings.simplefilter("ignore", SetuptoolsDeprecationWarning)
+    assert "License :: OSI Approved :: MIT License" in str(exc.value)
+
+
+def test_license_classifier_without_license_expression(tmp_path):
+    text = """\
+    [project]
+    name = "spam"
+    version = "2020.0.0"
+    license = {text = "mit or apache-2.0"}
+    classifiers = ["License :: OSI Approved :: MIT License"]
+    """
+    pyproject = _pep621_example_project(tmp_path, "README", text)
+
+    msg = "License classifiers are deprecated(?:.|\n)*MIT License"
+    with pytest.warns(SetuptoolsDeprecationWarning, match=msg):
         dist = pyprojecttoml.apply_configuration(makedist(tmp_path), pyproject)
-        # Check license classifier is still included
-        assert dist.metadata.get_classifiers() == [
-            "Development Status :: 5 - Production/Stable",
-            "Programming Language :: Python",
-            "License :: OSI Approved :: MIT License",
-        ]
+
+    # Check license classifier is still included
+    assert dist.metadata.get_classifiers() == ["License :: OSI Approved :: MIT License"]
```

Comment:
Missed that this uses `pytest.warns` instead of `pytest.raises`. So my previous post here was wrong. The nested block is actually fully executed.

Anyway, the current setup will also work.